src/player/PlayerEquipment.gd

```diff
@@ -31,9 +34,34 @@ func _ready() -> void:
 func get_inventory() -> Inventory:
 	return _inventory;
 
+
+func _on_inventory_inventory_event(item_stack, cursur_position, ev) -> void:
+	if ev is InputEventMouseButton && ev.pressed && ev.button_index == MOUSE_BUTTON_LEFT:
+		var hovered_rect: Rect2i;
+		
+		if item_stack == null:
+			hovered_rect = Rect2i(cursur_position, Vector2i.ONE);
+		else:
+			hovered_rect = Rect2i(item_stack.position, item_stack.get_rotated_size());
+		
+		var selection: InventoryRenderer.ItemSelection = _inventory_renderer.get_selection(ID_SELECTION)
+		if selection != null and selection.get_rect() == Rect2i(hovered_rect.position, hovered_rect.size):
```

Comment:
things outsidee of the inventory renderer shouldn't be aware of ItemSelection. the item stack has all the data you need in order to tell the renderer how to render the selection box